code/game/objects/structures/popout_cake.dm

```diff
@@ -159,3 +175,18 @@
 	slices_amount -= Clamp(round(Proj.damage / 3), 1, 4)
 
 	check_slices()
+
+//When spawned, stuffs the corpse underneath it inside
+/obj/structure/popout_cake/corpse_grabber/New()
+	..()
+
+	initialize()
```

Comment:
Actually now this is still gonna get called twice when the mapping SS initializes before objects, and objects calls initialize again.

Honk.

Welcome to SS13's broken init stack.